fdbbackup/FileDecoder.actor.cpp

```diff
@@ -517,57 +627,52 @@ class DecodeProgress {
 	int lfd = -1; // local file descriptor
 };
 
-ACTOR Future<Void> process_file(Reference<IBackupContainer> container, LogFile file, UID uid, DecodeParams params) {
+ACTOR Future<Void> process_file(Reference<IBackupContainer> container,
+                                LogFile file,
+                                UID uid,
+                                Reference<DecodeParams> params) {
 	if (file.fileSize == 0) {
 		TraceEvent("SkipEmptyFile", uid).detail("Name", file.fileName);
 		return Void();
 	}
 
-	state DecodeProgress progress(file, params.save_file_locally);
+	state DecodeProgress progress(file, params->save_file_locally);
 	wait(progress.openFile(container));
 	while (true) {
 		auto batch = progress.getNextBatch();
 		if (!batch.present())
 			break;
 
 		const VersionedMutations& vms = batch.get();
-		if (vms.version < params.beginVersionFilter || vms.version >= params.endVersionFilter) {
+		if (vms.version < params->beginVersionFilter || vms.version >= params->endVersionFilter) {
 			TraceEvent("SkipVersion").detail("Version", vms.version);
 			continue;
 		}
 
 		int sub = 0;
 		for (const auto& m : vms.mutations) {
 			sub++; // sub sequence number starts at 1
-			bool print = params.prefix.empty(); // no filtering
+			bool print = params->prefixes.empty(); // no filtering
```

Comment:
Nit. Can we just do

bool print = false;
if (!params->prefixes.empty()) {
  ...
}


The current code is confusing to read...